Enabling debug logger causes pytest memory leak that fails tests using make_napari_viewer

### 🐛 Bug Report

This issue was raised for cellfinder https://github.com/brainglobe/cellfinder/issues/443 but I'm able to reproduce it in a clean napari only environment.

When using `make_napari_viewer` combined with logging level set to debug, pytest will fail teardown because of a claimed qt memory leak.

### 💡 Steps to Reproduce

Create the python environment as below. Create a python file with this code:

```py
import numpy as np
import logging

import pytest


@pytest.mark.parametrize("debug_level", ["INFO", "WARNING", "ERROR", "DEBUG",])
def test_example(make_napari_viewer, debug_level):
    logger = logging.getLogger()
    logger.setLevel(getattr(logging, debug_level))

    viewer = make_napari_viewer()
    test_image = np.random.random((100, 100, 100))
    n_layers = len(viewer.layers)
    viewer.add_image(test_image, name="test_image")

    assert len(viewer.layers) == n_layers + 1
```

Run it with `pytest scratch.py --save-leaked-object-graph` to get the error.

### 💡 Expected Behavior

The pytest test should not fail. Or there should be a way to disable checking for these leaks if they don't affect the tests?

### 🌎 Environment

I created a virtualenv with python 3.10. Then did `pip install pytest napari npe2 pytest-qt pyqt5 graphviz objgraph`.

My `pip list` is:

```
Package                       Version
----------------------------- -----------
alabaster                     1.0.0
annotated-types               0.7.0
app-model                     0.4.0
appdirs                       1.4.4
asttokens                     3.0.0
attrs                         25.3.0
babel                         2.17.0
build                         1.3.0
cachey                        0.2.1
certifi                       2025.8.3
charset-normalizer            3.4.3
click                         8.3.0
cloudpickle                   3.1.1
colorama                      0.4.6
comm                          0.2.3
dask                          2025.9.1
debugpy                       1.8.17
decorator                     5.2.1
docstring_parser              0.17.0
docutils                      0.21.2
exceptiongroup                1.3.0
executing                     2.2.1
flexcache                     0.3
flexparser                    0.4
freetype-py                   2.5.1
fsspec                        2025.9.0
graphviz                      0.21
HeapDict                      1.0.1
hsluv                         5.0.4
idna                          3.10
imageio                       2.37.0
imagesize                     1.4.1
importlib_metadata            8.7.0
in-n-out                      0.2.1
iniconfig                     2.1.0
ipykernel                     6.30.1
ipython                       8.37.0
ipython_pygments_lexers       1.1.1
jedi                          0.19.2
Jinja2                        3.1.6
jsonschema                    4.25.1
jsonschema-specifications     2025.9.1
jupyter_client                8.6.3
jupyter_core                  5.8.1
kiwisolver                    1.4.9
lazy_loader                   0.4
locket                        1.0.0
magicgui                      0.10.1
markdown-it-py                4.0.0
MarkupSafe                    3.0.2
matplotlib-inline             0.1.7
mdurl                         0.1.2
napari                        0.6.4
napari-console                0.1.3
napari-plugin-engine          0.2.0
napari-svg                    0.2.1
nest-asyncio                  1.6.0
networkx                      3.4.2
npe2                          0.7.9
numpy                         2.2.6
numpydoc                      1.9.0
objgraph                      3.6.2
packaging                     25.0
pandas                        2.3.2
parso                         0.8.5
partd                         1.4.2
pillow                        11.3.0
Pint                          0.24.4
pip                           25.1.1
platformdirs                  4.4.0
pluggy                        1.6.0
pooch                         1.8.2
prompt_toolkit                3.0.52
psutil                        7.1.0
psygnal                       0.14.1
pure_eval                     0.2.3
pyconify                      0.2.1
pydantic                      2.11.9
pydantic-compat               0.1.2
pydantic_core                 2.33.2
Pygments                      2.19.2
PyOpenGL                      3.1.10
pyproject_hooks               1.2.0
PyQt5                         5.15.11
PyQt5-Qt5                     5.15.2
PyQt5_sip                     12.17.0
pytest                        8.4.2
pytest-qt                     4.5.0
python-dateutil               2.9.0.post0
pytz                          2025.2
pywin32                       311
PyYAML                        6.0.2
pyzmq                         27.1.0
qtconsole                     5.7.0
QtPy                          2.4.3
referencing                   0.36.2
requests                      2.32.5
rich                          14.1.0
rpds-py                       0.27.1
scikit-image                  0.25.2
scipy                         1.15.3
setuptools                    80.9.0
shellingham                   1.5.4
six                           1.17.0
snowballstemmer               3.0.1
Sphinx                        8.1.3
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
stack-data                    0.6.3
superqt                       0.7.6
tifffile                      2025.5.10
tomli                         2.2.1
tomli_w                       1.2.0
toolz                         1.0.0
tornado                       6.5.2
tqdm                          4.67.1
traitlets                     5.14.3
typer                         0.19.1
typing_extensions             4.15.0
typing-inspection             0.4.1
tzdata                        2025.2
urllib3                       2.5.0
vispy                         0.15.2
wcwidth                       0.2.13
wheel                         0.45.1
wrapt                         1.17.3
zipp                          3.23.0
```

### 💡 Additional Context

Likely there's some internal logging in napari that includes some objects in the logging context that gets stored somewhere globally, that contains a reference to qt object. But I could not comprehend the graphviz graphs to see who's the culprit.

Here's the generated graphviz graph: [QtViewer-leak-backref-graph-1.pdf](https://github.com/user-attachments/files/22443920/QtViewer-leak-backref-graph-1.pdf). And the original dot file (renamed to .txt): [objgraph-bxkr43er.txt](https://github.com/user-attachments/files/22443928/objgraph-bxkr43er.txt).

The full log output is:

```
$ pytest "D:\code_data\me\scratch.py" --save-leaked-object-graph
============================================================================================================ test session starts ============================================================================================================= platform win32 -- Python 3.10.11, pytest-8.4.2, pluggy-1.6.0
PyQt5 5.15.11 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: D:\code_data\me
plugins: napari-0.6.4, napari-plugin-engine-0.2.0, npe2-0.7.9, qt-4.5.0
collected 4 items

scratch.py ....E                                                                                                                                                                                                                        [100%]

=================================================================================================================== ERRORS =================================================================================================================== __________________________________________________________________________________________________ ERROR at teardown of test_example[DEBUG] __________________________________________________________________________________________________

item = <Function test_example[DEBUG]>

    @pytest.hookimpl(wrapper=True, trylast=True)
    def pytest_runtest_teardown(item):
        """
        Hook called after each test tear down, to process any pending events and
        avoiding leaking events to the next test. Also, if exceptions have
        been captured during fixtures teardown, fail the test.
        """
        _process_events()
        _close_widgets(item)
        _process_events()
>       result = yield

mem\lib\site-packages\pytestqt\plugin.py:205:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
mem\lib\site-packages\napari\utils\_testsupport.py:361: in make_napari_viewer
    fail_obj_graph(QtViewer)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

Klass = <class 'napari._qt.qt_viewer.QtViewer'>

    def fail_obj_graph(Klass):  # pragma: no cover
        """
        Fail is a given class _instances weakset is non empty and print the object graph.
        """

        try:
            import objgraph
        except ModuleNotFoundError:
            return

        if len(Klass._instances) != 0:
            global COUNTER
            COUNTER += 1
            import gc

            leaked_objects_count = len(Klass._instances)

            gc.collect()
            file_path = Path(
                f'{Klass.__name__}-leak-backref-graph-{COUNTER}.pdf'
            ).absolute()
            objgraph.show_backrefs(
                list(Klass._instances),
                max_depth=20,
                filename=str(file_path),
            )

            Klass._instances.clear()

            assert file_path.exists()

            # DO not remove len, this can break as C++ obj are gone, but python objects
            # still hang around and _repr_ would crash.
>           pytest.fail(
                f'Test run fail with leaked {leaked_objects_count} instances of {Klass}.'
                f'The object graph is saved in {file_path}.'
                f'{len(Klass._instances)} objects left after cleanup'
            )
E           Failed: Test run fail with leaked 1 instances of <class 'napari._qt.qt_viewer.QtViewer'>.The object graph is saved in D:\code_data\me\QtViewer-leak-backref-graph-1.pdf.0 objects left after cleanup

mem\lib\site-packages\napari\utils\_testsupport.py:78: Failed
------------------------------------------------------------------------------------------------------------- Captured log call -------------------------------------------------------------------------------------------------------------- DEBUG    in_n_out:_store.py:1091 Registering provider of <class 'napari.viewer.Viewer'>: <function _provide_viewer at 0x00000215B543A710> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering provider of <class 'napari.components.viewer_model.ViewerModel'>: <function _provide_viewer_model at 0x00000215B543A7A0> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering provider of <class 'napari._qt.qt_viewer.QtViewer'>: <function _provide_qt_viewer at 0x00000215B543A8C0> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering provider of <class 'napari._qt.qt_main_window.Window'>: <function _provide_window at 0x00000215B543A9E0> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering provider of <class 'napari.layers.base.base.Layer'>: <function _provide_active_layer at 0x00000215B543AB00> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering provider of <class 'napari.components.layerlist.LayerList'>: <function _provide_active_layer_list at 0x00000215B543AB90> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering processor of tuple[magicgui.widgets._function_gui.FunctionGui | PyQt5.QtWidgets.QWidget | magicgui.widgets.bases._widget.Widget, str]: <function _add_plugin_dock_widget at 0x00000215B5439FC0> (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.LayerDataTuple: <function _add_layer_data_tuples_to_viewer at 0x00000215B543A680> (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of list[napari.types.LayerDataTuple]: <function _add_layer_data_tuples_to_viewer at 0x00000215B543A680> (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of <class 'napari.layers.base.base.Layer'>: <function _add_layer_to_viewer at 0x00000215B543ACB0> (weight: 0, subclassable: True)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.ImageData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.ImageData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.ImageData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.ImageData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.LabelsData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.LabelsData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.LabelsData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.LabelsData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.PointsData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.PointsData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.PointsData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.PointsData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.ShapesData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.ShapesData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.ShapesData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.ShapesData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.SurfaceData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.SurfaceData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.SurfaceData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.SurfaceData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.TracksData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.TracksData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.TracksData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.TracksData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of napari.types.VectorsData: functools.partial(<function _add_layer_data_to_viewer at 0x00000215B543AC20>, return_type=napari.types.VectorsData) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:1091 Registering processor of concurrent.futures._base.Future[napari.types.VectorsData]: functools.partial(<function _add_future_data at 0x00000215B543AD40>, return_type=napari.types.VectorsData, _from_tuple=False) (weight: 0, subclassable: False)
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:953 Rebuilding provider map cache
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected ViewerModelToggleAction.__init__.<locals>.get_current(viewer: napari.components.viewer_model.ViewerModel) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting viewer: <class 'napari.components.viewer_model.ViewerModel'> = Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})
DEBUG    in_n_out:_store.py:797   Calling ViewerModelToggleAction.__init__.<locals>.get_current with {'viewer': Viewer(camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, mouse_pan=True, mouse_zoom=True, orientation=(<DepthAxisOrientation.TOWARDS: 'towards'>, <VerticalAxisOrientation.DOWN: 'down'>, <HorizontalAxisOrientation.RIGHT: 'right'>)), cursor=Cursor(position=(1.0, 1.0), viewbox=None, scaled=True, style=<CursorStyle.STANDARD: 'standard'>, size=1.0), dims=Dims(ndim=2, ndisplay=2, order=(0, 1), axis_labels=('0', '1'), rollable=(True, True), range=(RangeTuple(start=0.0, stop=2.0, step=1.0), RangeTuple(start=0.0, stop=2.0, step=1.0)), margin_left=(0.0, 0.0), margin_right=(0.0, 0.0), point=(np.float64(0.0), np.float64(0.0)), last_used=0), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False, spacing=0.0), layers=[], help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[], mouse_drag_callbacks=[<function drag_to_zoom at 0x00000215B52172E0>], mouse_double_click_callbacks=[<function double_click_to_zoom at 0x00000215B5217250>], mouse_wheel_callbacks=[<function dims_scroll at 0x00000215B52171C0>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, _keymap={})} (injected {'viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected _get_current_fullscreen_status(window: napari._qt.qt_main_window.Window) with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting window: <class 'napari._qt.qt_main_window.Window'> = <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>
DEBUG    in_n_out:_store.py:797   Calling _get_current_fullscreen_status with {'window': <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>} (injected {'window'})
DEBUG    in_n_out:_store.py:765 Executing @injected _get_current_menubar_status(window: napari._qt.qt_main_window.Window) with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting window: <class 'napari._qt.qt_main_window.Window'> = <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>
DEBUG    in_n_out:_store.py:797   Calling _get_current_menubar_status with {'window': <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>} (injected {'window'})
DEBUG    in_n_out:_store.py:765 Executing @injected _get_current_play_status(qt_viewer: napari._qt.qt_viewer.QtViewer) with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting qt_viewer: <class 'napari._qt.qt_viewer.QtViewer'> = <napari._qt.qt_viewer.QtViewer object at 0x00000215B5B6D870>
DEBUG    in_n_out:_store.py:797   Calling _get_current_play_status with {'qt_viewer': <napari._qt.qt_viewer.QtViewer object at 0x00000215B5B6D870>} (injected {'qt_viewer'})
DEBUG    in_n_out:_store.py:765 Executing @injected _get_current_tooltip_visibility() -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:797   Calling _get_current_tooltip_visibility with {} (injected set())
DEBUG    in_n_out:_store.py:765 Executing @injected _get_current_activity_dock_status(window: napari._qt.qt_main_window.Window) with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting window: <class 'napari._qt.qt_main_window.Window'> = <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>
DEBUG    in_n_out:_store.py:797   Calling _get_current_activity_dock_status with {'window': <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>} (injected {'window'})
DEBUG    in_n_out:_store.py:765 Executing @injected functools.partial(<function _get_current_dock_status at 0x00000215B58141F0>, full_name='Features table widget (napari builtins)')(*, full_name: str = 'Features table widget (napari builtins)') -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:797   Calling functools.partial(<function _get_current_dock_status at 0x00000215B58141F0>, full_name='Features table widget (napari builtins)') with {'full_name': 'Features table widget (napari builtins)'} (injected set())
DEBUG    in_n_out:_store.py:765 Executing @injected functools.partial(<function _get_current_dock_status at 0x00000215B58141F0>, full_name='Features table widget (napari builtins)')(*, full_name: str = 'Features table widget (napari builtins)') -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:797   Calling functools.partial(<function _get_current_dock_status at 0x00000215B58141F0>, full_name='Features table widget (napari builtins)') with {'full_name': 'Features table widget (napari builtins)'} (injected set())
DEBUG    in_n_out:_store.py:765 Executing @injected DockWidgetToggleAction.__init__.<locals>.get_current(window: napari._qt.qt_main_window.Window) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting window: <class 'napari._qt.qt_main_window.Window'> = <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>
DEBUG    in_n_out:_store.py:797   Calling DockWidgetToggleAction.__init__.<locals>.get_current with {'window': <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>} (injected {'window'})
DEBUG    in_n_out:_store.py:765 Executing @injected DockWidgetToggleAction.__init__.<locals>.get_current(window: napari._qt.qt_main_window.Window) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting window: <class 'napari._qt.qt_main_window.Window'> = <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>
DEBUG    in_n_out:_store.py:797   Calling DockWidgetToggleAction.__init__.<locals>.get_current with {'window': <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>} (injected {'window'})
DEBUG    in_n_out:_store.py:765 Executing @injected DockWidgetToggleAction.__init__.<locals>.get_current(window: napari._qt.qt_main_window.Window) -> bool with args: (), kwargs: {}
DEBUG    in_n_out:_store.py:787   injecting window: <class 'napari._qt.qt_main_window.Window'> = <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>
DEBUG    in_n_out:_store.py:797   Calling DockWidgetToggleAction.__init__.<locals>.get_current with {'window': <napari._qt.qt_main_window.Window object at 0x00000215B5ADA7A0>} (injected {'window'})
DEBUG    napari.layers.base.base:base.py:1513 Layer.refresh: test_image
DEBUG    napari.layers.base.base:base.py:1538 Layer._refresh_sync: test_image
DEBUG    napari.components._layer_slicer:_layer_slicer.py:194 _LayerSlicer.submit: layers=[<Image layer 'test_image' at 0x215c25c2020>], dims=ndim=3 ndisplay=2 order=(0, 1, 2) axis_labels=('0', '1', '2') rollable=(True, True, True) range=(RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0))) margin_left=(0.0, 0.0, 0.0) margin_right=(0.0, 0.0, 0.0) point=(np.float64(0.0), np.float64(0.0), np.float64(0.0)) last_used=0, force=False
DEBUG    napari.components._layer_slicer:_layer_slicer.py:229 Sync slicing for test_image
DEBUG    napari.layers.base.base:base.py:1260 Layer._slice_dims: test_image, dims=ndim=3 ndisplay=2 order=(0, 1, 2) axis_labels=('0', '1', '2') rollable=(True, True, True) range=(RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0))) margin_left=(0.0, 0.0, 0.0) margin_right=(0.0, 0.0, 0.0) point=(np.float64(0.0), np.float64(0.0), np.float64(0.0)) last_used=0, force=False
DEBUG    napari.components._layer_slicer:_layer_slicer.py:194 _LayerSlicer.submit: layers=[<Image layer 'test_image' at 0x215c25c2020>], dims=ndim=3 ndisplay=2 order=(0, 1, 2) axis_labels=('0', '1', '2') rollable=(True, True, True) range=(RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0))) margin_left=(0.0, 0.0, 0.0) margin_right=(0.0, 0.0, 0.0) point=(np.float64(49.0), np.float64(49.0), np.float64(49.0)) last_used=0, force=False
DEBUG    napari.components._layer_slicer:_layer_slicer.py:229 Sync slicing for test_image
DEBUG    napari.layers.base.base:base.py:1260 Layer._slice_dims: test_image, dims=ndim=3 ndisplay=2 order=(0, 1, 2) axis_labels=('0', '1', '2') rollable=(True, True, True) range=(RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0))) margin_left=(0.0, 0.0, 0.0) margin_right=(0.0, 0.0, 0.0) point=(np.float64(49.0), np.float64(49.0), np.float64(49.0)) last_used=0, force=False
DEBUG    napari.layers.base.base:base.py:1538 Layer._refresh_sync: test_image
DEBUG    napari.components._layer_slicer:_layer_slicer.py:194 _LayerSlicer.submit: layers=[<Image layer 'test_image' at 0x215c25c2020>], dims=ndim=3 ndisplay=2 order=(0, 1, 2) axis_labels=('0', '1', '2') rollable=(True, True, True) range=(RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0))) margin_left=(0.0, 0.0, 0.0) margin_right=(0.0, 0.0, 0.0) point=(np.float64(49.0), np.float64(49.0), np.float64(49.0)) last_used=0, force=False
DEBUG    napari.components._layer_slicer:_layer_slicer.py:229 Sync slicing for test_image
DEBUG    napari.layers.base.base:base.py:1260 Layer._slice_dims: test_image, dims=ndim=3 ndisplay=2 order=(0, 1, 2) axis_labels=('0', '1', '2') rollable=(True, True, True) range=(RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0)), RangeTuple(start=np.float64(0.0), stop=np.float64(99.0), step=np.float64(1.0))) margin_left=(0.0, 0.0, 0.0) margin_right=(0.0, 0.0, 0.0) point=(np.float64(49.0), np.float64(49.0), np.float64(49.0)) last_used=0, force=False
---------------------------------------------------------------------------------------------------------- Captured stdout teardown ---------------------------------------------------------------------------------------------------------- Graph written to C:\Prog\msys64\tmp\objgraph-bxkr43er.dot (407 nodes)
Image generated as D:\code_data\me\QtViewer-leak-backref-graph-1.pdf
----------------------------------------------------------------------------------------------------------- Captured log teardown ------------------------------------------------------------------------------------------------------------ DEBUG    napari.components._layer_slicer:_layer_slicer.py:259 _LayerSlicer.shutdown
========================================================================================================== short test summary info =========================================================================================================== ERROR scratch.py::test_example[DEBUG] - Failed: Test run fail with leaked 1 instances of <class 'napari._qt.qt_viewer.QtViewer'>.The object graph is saved in D:\code_data\me\QtViewer-leak-backref-graph-1.pdf.0 objects left after cleanup
======================================================================================================== 4 passed, 1 error in 16.71s =========================================================================================================
```

--- a/src/napari/utils/_testsupport.py
+++ b/src/napari/utils/_testsupport.py
@@ -1,4 +1,5 @@
 import gc
+import logging
 import os
 import sys
 import warnings
@@ -456,3 +457,29 @@ def MouseEvent():
         handled: bool = False
 
     return Event
+
+
+class LeakSafeLogRecord(logging.LogRecord):
+    """LogRecord that converts args to strings to prevent reference retention."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Convert args to strings immediately
+        if self.args:
+            self.args = tuple(
+                str(arg) if not isinstance(arg, int | float) else arg
+                for arg in self.args
+            )
+
+
+@pytest.fixture(autouse=True, scope='session')
+def use_leak_safe_log_records():
+    """Use custom LogRecord factory that doesn't retain object references."""
+    original_record_factory = logging.getLogRecordFactory()
+
+    logging.setLogRecordFactory(LeakSafeLogRecord)
+
+    yield
+
+    # Restore original factory
+    logging.setLogRecordFactory(original_record_factory)

